server/plugins/builder/pages/index.js

```diff
@@ -19,39 +19,49 @@ class Page extends EngineBasePage {
     return {
       payload: {
         output: 'stream',
-        parse: true
+        parse: true,
+        maxBytes: 5e+6 // 5mb
       },
       ext: {
         onPreHandler: {
           method: async (request, h) => {
             let files = fileStreamsFromPayload(request.payload)
-            if (!files.length) {
-              return h.continue
-            }
-            let file = files[0]
-            let key = file[0]
-            let previousUpload = (request.yar.get('originalFilenames') || {})[key]
-            if (previousUpload && file[1].hapi.filename === '') {
-              h.request.payload[key] = previousUpload.location
-              return h.continue
-            }
-            try {
-              // TODO:- should be for each(?) or limit to one upload per request..? 🤔
-              let saved = await saveFileToTmp(file[1])
-              let { error, location } = await uploadDocument(saved)
-              if (location) {
-                request.yar.set('originalFilenames', { [key]: { originalFilename: file[1].hapi.filename, location } })
-                h.request.payload[key] = location
+            let state = await this.model.getState(request)
+            let originalFilenames = state.originalFilenames || {}
+
+            files.forEach(file => {
+              let key = file[0]
+              let previousUpload = originalFilenames[key]
+              if (previousUpload && file[1]._data.length) {
+                h.request.payload[key] = previousUpload.location
+              } else {
+                delete request.payload[key]
               }
-              if (error) {
-                h.request.pre.errors = [{
-                  path: key, href: `#${key}`, name: key, text: 'This file contains a virus'
-                }]
+            })
+
+            if (!files.length || files.find(file => file[1]._data.length < 1)) {
+              return h.continue
+            } else {
+              let file = files[0]
+              let key = file[0]
+              try {
+                // TODO:- should be for each(?) or limit to one upload per request..? 🤔
```

Comment:
We should pick one, at the moment some stuff deals with mulitple files (e.g. verifying atleast one uploadeed file has data) and some stuff doesn't, e.g. this getting the first file. I think let's add a foreach and handle multiple all around.

Reply:
if there are multiple uploads on a page, should we upload only when all fields are populated or upload what we can and tell the applicant some are empty?

Reply:
I guess if we don't upload the valid ones they'll have to re-select them in the page? So let's upload what's valid.